libpdf/textbox.py

```diff
@@ -214,12 +215,15 @@ def render_chapters(  # pylint: disable=too-many-branches, too-many-locals
                         lt_textboxes.remove(lt_textbox)
 
                 else:
-                    # if no matched lt_textboxes are found, the position of a virtual chapter is rendered
-                    position = virtual_chapter_position_generator(chapter, chapter_page)
+                    # if no matched lt_textboxes are found for the chapter in the outline, a ghost chapter is rendered.
+                    # The ghost chapter contains only the position from the outline and have no textbox
+                    position = ghost_chapter_position_generator(chapter, chapter_page)
                     horizontal_box = None
                     LOG.info(
-                        'The chapter "%s" on page %s can not be detected. The virtual chapter number "%s" is applied. '
-                        'This number may not be consistent with the numerical order in the content',
+                        'The chapter "%s" on page %s can not be detected. the position of a ghost chapter is rendered. '
+                        'The chapter number "%s" is applied. '
+                        'This number may not be consistent with the numerical order in the content '
+                        'as a virtual number may be introduced.',
```

Comment:
So the `virtual` side of things is completely independent. How can you know this here if you don't check for it?